### PR TITLE
added `new` command

### LINKED ISSUE #118: [slight] impl new command to create new slight project
**Describe the solution you'd like**
`slight new -t rs -n <some_name>` should create a rust project with slight dependencies. 

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by danbugs**
closes #118 

here is an example of creating a new c slight project and running it:
image

here is an example of creating a new rust slight project and running it:
image

Signed-off-by: Dan Chiarlone <dchiarlone@microsoft.com>

### CODE REVIEW

**Comment by Mossaka** [comment]
Does this work in Win?

**Comment by danbugs** [comment]
> Does this work in Win?

@Mossaka ~ Yes! I've tested it on Windows and it works fine. Although, I'll probably have to adapt the `bindings` rule 'cause of `chmod`.

File: slight/Cargo.toml [hunk 1]
```diff
@@ -35,6 +35,8 @@ as-any = "0.3"
 tracing-subscriber = { version = "0.3", features = ["env-filter"] }
 tracing = { version = "0.1", features = ["log"] }
 reqwest = "0.11"
+flate2 = "1"
+tar = "0.4"
```
[the remark this answers is not in the document]
Comment (danbugs) [reply slight/Cargo.toml:39]: No

File: Makefile [hunk 2]
```diff
@@ -145,4 +145,14 @@ build-app-demos:
...
\ No newline at end of file
+### END OF APP DEMO
+
+### GITHUB RELEASES
+.PHONY: release
+release:
+	tar -czvf slight.tar.gz target/release/slight
+	tar -C templates/ -czvf rust-template.tar.gz rust
+	tar -C templates/ -czvf c-template.tar.gz c
+	gh release create $(TAG) --generate-notes -p slight.tar.gz wit/configs.wit wit/event-handler.wit wit/events.wit wit/http-handler.wit wit/http-types.wit wit/http.wit wit/kv.wit wit/mq.wit wit/pubsub.wit wit/resources.wit wit/types.wit rust-template.tar.gz c-template.tar.gz
```
[the remark this answers is not in the document]
Comment (danbugs) [reply Makefile:156]: sure thing~ I have a new PR coming up w/ some changes and will update it there just to avoid having to re-run the pipelines here!

**Pull request merged**